code/modules/supply/supply_console.dm

```diff
@@ -0,0 +1,499 @@
+		if(order.ordered_by_department.crate_auto_approve && order.ordered_by_department.auto_approval_cap >= order.object.cost)
+			approve_crate(user, order.ordernum)
+		investigate_log("| [key_name(user)] has placed an order for [order.object.amount] [order.object.name] with reason: '[order.comment]'", "cargo")
+
+/obj/machinery/computer/supplycomp/proc/approve_crate(mob/user, order_num)
+	for(var/datum/supply_order/department_order in SSeconomy.request_list)
+		if(department_order.ordernum == order_num)
+			var/datum/supply_order/order = department_order
+			var/datum/supply_packs/pack = order.object
+			var/datum/money_account/account = order.orderedbyaccount
+
+			if(order.requires_qm_approval)
+				if(!has_qm_access(user.get_access()))
+					return FALSE
+				order.requires_qm_approval = FALSE
+				if(account.account_type == ACCOUNT_TYPE_PERSONAL || isnull(order.ordered_by_department))
+					if(pay_with_account(account, order.object.cost, "[pack.name] Crate Purchase", "Cargo Requests Console", user, account_database.vendor_account))
+						SSeconomy.process_supply_order(order, TRUE) //send 'er back through
+						return TRUE
+					atom_say("ERROR: Account tied to order cannot pay, auto-denying order")
+					SSeconomy.request_list -= order //just remove order at this poin
+				else
+					return TRUE
+				return TRUE
+
+			if(order.requires_head_approval)
+				//if they do not have access to this account
+				if(!department_order.ordered_by_department.has_account_access(user.get_access(), user.get_worn_id_account()))
+					//and the dept account doesn't have auto approve enabled (or does and the crate is too expensive for auto approve)
+					if(!department_order.ordered_by_department.crate_auto_approve || department_order.ordered_by_department.auto_approval_cap < pack.cost)
+						return //no access!
+
+				///just give the account pin here, its too much work for players to get the department account pin number since approval is access locked anyway
+				if(attempt_account_authentification(account, user, account.account_pin))
+					if(pay_with_account(account, pack.cost, "[pack.name] Crate Purchase", "[src]", user, account_database.vendor_account))
+						order.requires_head_approval = FALSE
+						SSeconomy.process_supply_order(order, TRUE)
+						investigate_log("| [key_name(user)] has authorized an order for [pack.name]. Remaining Cargo Balance: [cargo_account.credit_balance].", "cargo")
+						SSblackbox.record_feedback("tally", "cargo_shuttle_order", 1, pack.name)
+					else
+						atom_say("ERROR: Account tied to order cannot pay, auto-denying order")
+						SSeconomy.request_list -= order
+
+
+			//how did we get here?
+
+/obj/machinery/computer/supplycomp/proc/deny_crate(mob/user, order_num)
+	var/obj/item/card/id/C
+	if(ishuman(user))
+		var/mob/living/carbon/human/H = user
+		C = H.get_idcard(TRUE)
+	for(var/datum/supply_order/department_order as anything in SSeconomy.request_list)
+		if(department_order.ordernum == order_num)
```

Comment:
Same here. Early continues are preferred to keep the indentation level down

Reply:
done in next commit